packages/amplify_api/lib/src/graphql/graphql_request_factory.dart

```diff
@@ -101,12 +101,19 @@ class GraphQLRequestFactory {
   }
 
   DocumentInputs _buildDocumentInputs(
-      ModelSchema schema, GraphQLRequestOperation operation) {
+      ModelSchema schema, GraphQLRequestOperation operation,
+      [Map<String, dynamic>? documentArgs]) {
     String upperOutput = '';
     String lowerOutput = '';
     List<String> upperList = [];
     List<String> lowerList = [];
 
+    if (documentArgs != null) {
+      documentArgs.forEach((key, value) {
+        lowerList.add("$key: $value");
+      });
+    }
```

Comment:
Let's remove this and use what you have for the `default` impl below. Ideally, the document we're building here needs no knowledge of the variables.

Reply:
+1 . I'm a little confused how the `documentArgs` will support this (the pagination parameters) as well as query params and other variables. When we add the query params, it might make it a little easier to handle this param correctly. If you wanted, you could even remove the whole `documentArgs` parameter in this PR and just pass the limit as a primitive parameter very specific to list case, similar to the way you added id as primitive param in the get helper.

Reply:
Thanks for the clarification, this section has been refactored!